### DISCUSSION TITLE
Exit and restart on second launch

### DISCUSSION THREAD

**Discussion by scubanarc** [ndbeals/winssh-pageant#6] [Ideas]
I think that the current behavior is to only allow a single instance, but I don't quite see how in the code. At least it seems that when I launch it a second time it refuses to launch if another instance is running.

What do you think about exit/restart on second instance?

**Comment by ndbeals** [comment]
Apologies for the late reply, new github feature for me and I missed the notification.

The reason a second instance refuses to launch/work is two-fold:

1. Because it's listening to a named pipe `\\.\pipe\pageant.%s.%s` - New pageant behaviour (was in beta as of august 2020)
2. Because it's creating a specifically named and classed window - Old pageant behaviour (how it's worked since the start)

Both of those operations are exclusive (AFAIK, I'm not an expert at windows api). 


> What do you think about exit/restart on second instance?

By this, do you mean the first instance exits itself when a second instance is launched? or the second instance exits itself if one is already running? I'm not sure there's much utility in having it restart itself but I could be misunderstanding what you mean.
